src/analyzer.rs

```diff
@@ -214,21 +207,23 @@ impl<'tree> Traversable<'tree> for Analyzer {
     /// This methods collects treesitter nodes with BFS
     ///
     /// All of tree sitter nodes are ordered by non decreasing order
-    fn get_scannable_nodes(&self, tree: &'tree Tree) -> Vec<(Node<'tree>, (usize, usize, usize))> {
-        let mut deq = VecDeque::new();
+    fn get_scannable_nodes<'b>(&self, tree: &'tree Tree) -> Vec<(Node<'tree>, (usize, usize, usize))> {
```

Comment:
The while statement below cannot work because it is receiving a mutable reference to a VecDeque from `enqueue_child_nodes`.

If you pass ownership of the `VecDeque` to `enqueue_child_nodes` and pass it back to `VecDeque` as the return type, you should be able to work around this.

```rust
    fn get_scannable_nodes(&self, tree: &'tree Tree) -> Vec<(Node<'tree>, (usize, usize, usize))> {
// ...
        let mut deq: VecDeque<Node<'tree>> = VecDeque::new();
        deq.push_back(tree.root_node());
// ... in while
                deq = self.enqueue_child_nodes(deq, &node);
// ... return
        result
```rust

```rust
    fn enqueue_child_nodes(
        &self,
        mut deq: VecDeque<Node<'tree>>,
        node: &Node<'tree>,
    ) -> VecDeque<Node<'tree>> {
```